pipeline/src/handler.ts

```diff
@@ -20,34 +30,58 @@ export const createHandler =
       settings: articles.settings,
     });
 
-    // 1. Fetches everything from Prismic
+    // 1. Prepare queries to fetch everything from Prismic
     const window = toBoundedWindow(event);
+    log.info(`Fetching documents last published ${describeWindow(window)}`);
+
+    // 2. Fetch all documents published in the given window and partition them into
+    // "parent" and "other" documents: those which we want to transform and those
+    // which *may* have been denormalised onto the former.
+    const [parentDocuments, otherDocuments] = partition(
+      paginator((after?: string) =>
+        getPrismicDocuments(clients.prismic, {
+          publicationWindow: toBoundedWindow(event),
+          graphQuery,
+          after,
+        })
+      ),
+      isArticle
+    );
+
+    // 3. Index all the parent documents into elasticsearch after transforming them
+    // into our common schema
+    const initialIndex = await bulkIndexDocuments(
+      clients.elastic,
+      articles.index,
+      parentDocuments.pipe(map(transformArticle))
+    );
     log.info(
-      `Fetching articles and webcomics last published ${describeWindow(window)}`
+      `Indexed ${initialIndex.successfulIds.size} parent documents in ${initialIndex.time}ms`
     );
-    const getArticles = (after?: string) =>
-      getPrismicDocuments<ArticlePrismicDocument>(clients.prismic, {
-        graphQuery: wrapQueries(articlesQuery, webcomicsQuery),
-        contentTypes: ["articles", "webcomics"],
-        publicationWindow: toBoundedWindow(event),
-        after,
-      });
-
-    // 2. Transforms each thing
-    const articleObservable = from(getArticles()).pipe(
-      expand((res) => (res.lastDocId ? getArticles(res.lastDocId) : EMPTY)),
-      mergeMap((page) => page.docs),
-      map(transformArticle)
+
+    // 3. Find parent documents which were not changed but have child documents that
+    // were changed
+    const nextParentDocuments = otherDocuments.pipe(
+      map((doc) => doc.id),
+      // Query elasticsearch for (parent) documents that contain these child document IDs
+      // The field name is mapped in `indices/articles.ts` and populated by the transformer
+      getParentDocumentIDs(clients.elastic, {
+        index: articles.index,
+        identifiersField: "query.linkedIdentifiers",
+      }),
+      // We don't need to update parent documents that we already got in this window
+      // as they were updated above
+      filter((parentId) => !initialIndex.successfulIds.has(parentId)),
+      // Fetch the latest version of all the parent documents
+      getDocumentsByID<ArticlePrismicDocument>(clients.prismic, { graphQuery })
```

Comment:
IIUIC, this fetches the parent document with its child content denormalised into it, meaning that we don't have to do any fiddling after this to insert the child content.

Could something along those lines be added to the comment?  It would help with understanding both the why and the where. (Why don't we just get it from ES, since we're doing the IDs query, above?  Where do we populate the parent with the new data?)